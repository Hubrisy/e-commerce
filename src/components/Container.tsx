import type { PropsWithChildren } from 'react';
import clsx from 'clsx';

interface Props {
  className?: string;
}

export const Container: React.FC<PropsWithChildren<Props>> = ({
  children,
  className,
}) => (
  <div className={clsx('max-w-[1120px] mx-auto px-large', className)}>
    {children}
  </div>
);
