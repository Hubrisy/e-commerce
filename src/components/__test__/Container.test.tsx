import { render } from '@testing-library/react';

import { Container } from '../Container';

describe('Container', () => {
  test('renders correctly', () => {
    const Component = () => (
      <Container className="test">
        <div>Test children</div>
      </Container>
    );

    const { asFragment } = render(<Component />);
    expect(asFragment()).toMatchSnapshot();
  });
});
