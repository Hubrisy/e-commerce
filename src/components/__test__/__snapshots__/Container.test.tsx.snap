// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container renders correctly 1`] = `
<DocumentFragment>
  <div
    class="max-w-[1120px] mx-auto px-large test"
  >
    <div>
      Test children
    </div>
  </div>
</DocumentFragment>
`;
